### PR TITLE
Update front page

### PR DESCRIPTION
**Pull request by mattip**
Update the front page of numpy.org:

- remove strange iframe template code
- update scipy-sphinx-theme (to latest head)
- make the links reference the devdocs on this site
~~- add content about obtaining (starting from [here](https://www.scipy.org/scipylib/download.html), but updated, simplified, mention Intel MKL, and removed obsolete [install](https://www.scipy.org/install.html) page)~~
  - sync back to  scipy.org.scipylib?
- made links point to devdocs uploaded as part of the numpy build process via CircleCI
- added previous doc links (still point to scipy.org)
- made openhub badge show up again

The donate button points to FlipCause.com, I think that is correct but waiting for confimation from NumFocus

### CODE REVIEW

File: www/index.rst [hunk 1]
```diff
@@ -28,24 +29,48 @@ few restrictions.
...
 
-- `Getting NumPy <http://www.scipy.org/scipylib/download.html>`__
-- `Installing the SciPy Stack <http://www.scipy.org/install.html>`__
-- `NumPy and SciPy documentation page <http://docs.scipy.org/doc/>`__
-- `NumPy Tutorial <https://docs.scipy.org/doc/numpy/user/quickstart.html>`__
-- `NumPy for MATLAB© Users <https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html>`__
-- `NumPy functions by category <https://docs.scipy.org/doc/numpy/reference/routines.html>`__
-- `NumPy Mailing List <http://www.scipy.org/scipylib/mailing-lists.html>`__
+- :ref:`obtaining`
+- `NumPy Tutorial <devdocs/user/quickstart.html>`__
```
Comment (rgommers) [inline www/index.rst]: Can we keep these `devdocs` links as absolute links instead? The devdocs aren't part of this repo, and as relative links they're impossible to verify before deployment with a local build of the site.

File: www/index.rst [hunk 2]
```diff
@@ -28,24 +29,48 @@ few restrictions.
...
+
+Documentation
+-------------
+
+`Latest (development) version <devdocs>`__
+
+* `Version 1.15.4 <https://docs.scipy.org/doc/numpy-1.15.4>`__,
+  `[HTML+zip] <https://docs.scipy.org/doc/numpy-1.15.4/numpy-html-1.15.4.zip>`__
+  `[PDF] <https://docs.scipy.org/doc/numpy-1.15.4/numpy-ref-1.15.4.pdf>`__
+* `Version 1.15.1 <https://docs.scipy.org/doc/numpy-1.15.1>`__,
```
Comment (rgommers) [inline www/index.rst]: I suggest only to put the `.0` and the `.most-recent` links, the rest is just noise.

File: www/obtaining.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
+.. _obtaining:
+
+===============
+Obtaining NumPy
+===============
```
Comment (rgommers) [inline www/obtaining.rst]: I'm not convinced about adding this page. It is in general not the right advice for beginning users, which are the main set of users that will follow the `obtaining` link probably. I'd much rather just link to https://scipy.org/install.html.

File: www/index.rst (hunk 2 above)
Comment (rgommers) [reply www/index.rst]: oh yes, and can you add 1.16.x?

File: www/index.rst (hunk 1 above)
Comment (mattip) [reply www/index.rst]: ok

File: www/index.rst (hunk 2 above)
Comment (mattip) [reply www/index.rst]: I will remove the noise. One of the goals of this is to locally host the 1.16.x docs on numpy.org, so I would like to add them as a second PR once we have a way to upload them there

File: www/obtaining.rst (hunk 3 above)
Comment (mattip) [reply www/obtaining.rst]: This is meant to eventually replace that outdated information with our current "best practices". What is not the right advice?

File: www/index.rst (hunk 2 above)
Comment (mattip) [reply www/index.rst]: The `.0` versions or the latest in the series (1.14.5, 1.15.4, ...)?

File: www/obtaining.rst (hunk 3 above)
Comment (mattip) [reply www/obtaining.rst]: ok, backing this out

File: www/index.rst (hunk 2 above)
Comment (mattip) [reply www/index.rst]: Now reduced to 1.16.2, 1.15.4, 1.14.5, 1.13.0

File: www/index.rst [hunk 4]
```diff
@@ -28,24 +29,36 @@ few restrictions.
...
 .. raw:: html
 
    <div style="padding-top:40px; text-align:center; margin:0px auto;">
      <div style="display: inline-block; float:left; margin-left:1em;">
        <a href ="https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==" target="_blank" style="background:#f5cb33; border-radius:0px 0px 0px 0px;font-weight:normal; font-family:Arial, Helvetica, sans-serif;border:none;box-shadow:none; clear: both;display: block; width:150px;height:34px; line-height:2.8; font-size:13px;text-align:center; cursor:pointer;color:#000000;text-decoration: none; z-index:1">Donate to Numpy</a>
      </div>
-     <div style="display: inline-block;">
-       <script type="text/javascript" src="https://www.ohloh.net/p/4894/widgets/project_partner_badge.js"></script>
-     </div>
+     <script type='text/javascript' src='https://www.openhub.net/p/numpy/widgets/project_partner_badge?format=js'></script>
```
Comment (mattip) [inline www/index.rst]: No, this does the wrong thing. It displays the form directly rather than adding a "Donate" button.

**Comment by mattip** [comment]
sorry for all the commits. This should be merge-squashed.

**Comment by rgommers** [comment]
I get
```
Your connection is not secure

The owner of numpy.org has configured their website improperly. To protect your information from being stolen, Firefox has not connected to this website.

The certificate is only valid for the following names: www.github.com, *.github.io, *.githubusercontent.com, *.github.com, github.com, github.io, githubusercontent.com
```
for https://numpy.org/devdocs at the moment. Anyone aware of us changing anything?

File: www/index.rst [hunk 5]
```diff
@@ -30,22 +30,38 @@ Getting Started
...
+
+
+Documentation
+-------------
+
+`Latest (development) version <https:/numpy.org/devdocs>`__
+
+* `Version 1.16.2 <https://docs.scipy.org/doc/numpy-1.16.2>__,
+  `[HTML+zip] <https://docs.scipy.org/doc/numpy-1.16.2/numpy-html-1.15.4.zip>`__
+  `[PDF] <https://docs.scipy.org/doc/numpy-1.16.2/numpy-ref-1.15.4.pdf>`__
```
Comment (rgommers) [inline www/index.rst]: these two 1.16.2 links are incorrect

**Comment by mattip** [comment]
Who manages numpy.org? Perhaps the certificate expired?

**Comment by stefanv** [comment]
On Thu, 21 Mar 2019 17:11:18 +0000, Ralf Gommers wrote:
> I get
> ```
> Your connection is not secure
> 
> The owner of numpy.org has configured their website improperly. To protect your information from being stolen, Firefox has not connected to this website.
> 
> The certificate is only valid for the following names: www.github.com, *.github.io, *.githubusercontent.com, *.github.com, github.com, github.io, githubusercontent.com
> ```
> for https://numpy.org/devdocs at the moment. Anyone aware of us
> changing anything?

It should redirect to https://www.numpy.org/devdocs/, which looks OK
from over here.

**Comment by mattip** [comment]
my bad - I used https://numpy.org not https://www.numpy.org

**Comment by mattip** [comment]
fixed the links, actually ran `make linkcheck` this time, and squashed the too-many commits into one.

**Comment by stefanv** [comment]
numpy.org lives on GitHub, so certificates should never be an issue.

**Comment by stefanv** [comment]
Any reason why we have the `www` in there?  Feels like `numpy.org` looks better.

**Comment by mattip** [comment]
running `make linkcheck` with `https://numpy.org` gave an error, with `https://www.numpy.org` did not give an error.

**Comment by stefanv** [comment]
Yes, `www.numpy.org` is the *de facto* site, so I'm not surprised.  But I would like to consider changing that; it may just be a bit tricky to make the transition smooth; I'd have to investigate HTTPS redirect (I think it is possible).

File: www/index.rst (hunk 2 above)
Comment (rgommers) [reply www/index.rst]: thanks, looks better

**Comment by rgommers** [comment]
Fixed a small typo, the rest all looked good. Merged, thanks @mattip

**Pull request merged**